engine/src/application.cpp

```diff
@@ -2,30 +2,38 @@
 
 // TODO: Replace this with a proper implementation
 
-#include "application.hpp"
+#include "gen/application.hpp"
 #include <numbers>
 
 namespace gen
 {
-	// possibly change this to instead use Update() and Draw() functions
 
-	void Application::run()
+	Application::Application(const char * appName, mim::vec2i const & initialSize) : m_engine{appName, initialSize}
 	{
-		gameLoop();
-		shutdown();
 	}
 
-	void Application::gameLoop()
+	Application::~Application()
+	{
```

Comment:
Empty destructor: pessimization, please use `= default` in the header, or call `shutdown()` here (which should wait for the device to be idle).